R/flatten_data.R

```diff
@@ -1,7 +1,42 @@
+#' determine data type
+#' 
+#' @param data (list or tbl_df, tbl, data.frame) An ecocomDP formatted dataset (list), or a list of ecocomDP tables, a single flat ecocomDP table, or a single flat table created by joinng and spreading ecocomDP tables. 
+#' 
+#' @return (character) A character string indicating data type. One of: 'list_of_tables', 'datset_old', 
+#' 'dataset', or 'flat_table'
+#' 
+#' @details Internal function for detecting datatype. Not exported
+detect_data_type <- function(data){
```

Comment:
@sokole This function needs unit tests.

Reply:
Here are some examples you could use in unit testing:

```
# detecting data types
ecocomDP:::detect_data_type(ants_L1)
ecocomDP:::detect_data_type(ants_L1[[1]])
ecocomDP:::detect_data_type(ants_L1[[1]]$tables)
ecocomDP:::detect_data_type(ants_L1[[1]]$tables$observation)

ecocomDP:::detect_data_type(my_dataset)
ecocomDP:::detect_data_type(my_dataset[[1]])
ecocomDP:::detect_data_type(my_dataset[[1]]$tables)
ecocomDP:::detect_data_type(my_dataset[[1]]$tables$observation)

# error out with informative message
ecocomDP:::detect_data_type(ants_L1[[1]]$metadata)
ecocomDP:::detect_data_type(my_dataset[[1]]$metadata)
```

Reply:
Nice @sokole, now put these into a unit test following the standard testthat procedures ([more on that here](https://r-pkgs.org/tests.html)).

Reply:
I'll handle this @sokole